dashboard/model_layer.py

```diff
@@ -39,29 +40,21 @@ def get_sessions_count_by_service_type(node_key, service_type):
 
 
 def get_average_session_time():
-    sessions = models.Session.query.all()
-
-    count = 0
-    total_seconds = 0
-
-    for se in sessions:
-        updated_at = se.node_updated_at or se.client_updated_at
-        if updated_at:
-            delta = updated_at - se.created_at
-            total_seconds += delta.total_seconds()
-            count += 1
-
-    average_in_seconds = total_seconds / count if count != 0 else 0
-    return helpers.format_duration(timedelta(seconds=average_in_seconds))
+    sql = text("""select
+    AVG(TIME_TO_SEC(TIMEDIFF(client_updated_at, created_at)))
+    as averageDuration FROM session;""")
```

Comment:
I think it's possible to do these queries with SQLAlchemy, but not sure if it's easy, so I'm fine with SQL :D